### ISSUE TITLE
Allow arbitrary data to be passed into Toasts

### ISSUE THREAD

**Issue by JamesPerlman** [calintamas/react-native-toast-message#79]
This would be super useful for custom UI's that need more than 2 text fields to configure (like an onPress handler, for example).

Proposal:

A custom toast component:
```
const MyInfoToast = (props) => (
  <TouchableOpacity onPress={props.onPress}>
    <Text>{'I am a Touchable Toast!'}</Text>
  </TouchableOpacity>
);
```

Toast declaration with config:
```
<Toast
  ref={(ref) => Toast.setRef(ref)}
  config={{
    info: ({ props }) => <MyInfoToast {...props} />,
  }}
/>
```

Showing Toast with props:
```
Toast.show({ type: 'info', props: { onPress: () => console.log('i got pressed!') } });
```

This could be easily achieved by just adding the value 'props' to the `keys` array starting at line 263 of `src/index.js`

Curious to know what you think of this!

Thanks for making this repo by the way, it's really great :)

**Comment by troyvnit** [comment]
I created the PR to resolve this issue here https://github.com/calintamas/react-native-toast-message/pull/81 . Please review and merge if you think it's useful.

**Comment by calintamas** [comment]
published @troyvnit 's changes in `v1.3.5`
